src/main/java/org/matsim/run/prepare/PreparePopulation.java

```diff
@@ -11,45 +11,96 @@
+
+    @CommandLine.Parameters(arity = "1", paramLabel = "INPUT", description = "Path to input population")
+    private Path input;
+
+    @CommandLine.Option(names = "--output", description = "Path to output population", required = true)
+    private Path output;
+
+    @Override
+    public Integer call() throws Exception {
+
+        if (!Files.exists(input)) {
+            log.error("Input population does not exist: {}", input);
+            return 2;
+        }
+
+        Population population = PopulationUtils.readPopulation(input.toString());
+
+        for (Person person : population.getPersons().values()) {
+            // Set car availability to "never" for agents below 18 years old
+            Object age = person.getAttributes().getAttribute("microm:modeled:age");
+            String avail = "always";
+            if (age != null && (int) age < 18)
+                avail = "never";
+            PersonUtils.setCarAvail(person, avail);
+
+            // Assign income to person
+            String incomeGroupString = (String) person.getAttributes().getAttribute("MiD:hheink_gr2");
+            String householdSizeString = (String) person.getAttributes().getAttribute("MiD:hhgr_gr");
+            int incomeGroup = 0;
+            double householdSize = 1;
+            if (incomeGroupString != null && householdSizeString != null) {
+                incomeGroup = Integer.parseInt(incomeGroupString);
+                householdSize = Double.parseDouble(householdSizeString);
+            }
+
+            double income = 0;
+            Random rnd = new Random(1234);
+            switch (incomeGroup) {
+                case 1:
+                    income = 500 / householdSize;
+                    break;
+                case 2:
+                    income = (rnd.nextInt(400) + 500) / householdSize;
+                    break;
+                case 3:
+                    income = (rnd.nextInt(600) + 900) / householdSize;
+                    break;
+                case 4:
+                    income = (rnd.nextInt(500) + 1500) / householdSize;
+                    break;
+                case 5:
+                    income = (rnd.nextInt(1000) + 2000) / householdSize;
+                    break;
+                case 6:
+                    income = (rnd.nextInt(1000) + 3000) / householdSize;
+                    break;
+                case 7:
+                    income = (rnd.nextInt(1000) + 4000) / householdSize;
+                    break;
+                case 8:
+                    income = (rnd.nextInt(1000) + 5000) / householdSize;
+                    break;
+                case 9:
+                    income = (rnd.nextInt(1000) + 6000) / householdSize;
+                    break;
+                case 10:
+                    income = (rnd.nextInt(3000) + 7000) / householdSize;
+                    break;
+                default:
+                    income = 2389.0; // Average monthly household income per Capita (2018)
```

Comment:
Can you add a reference (in comments) ? Thanks
I was a little confused by "household income per capita". Is it the average per capita or per household?

Reply:
just saw the reference in the PR comment. I think it would be good to also have it in the code. Furthermore, i think it would be good tu use a German database (like statista) -because otherwise we rely on $ -> € conversion rate. In the end, i'd also say that, if attributes do not provide enough information, we take the average income of a person.
Alternatively/Additionally, if e.g. only household size is provided, we could divide the average household income by the given household size. But that is maybe less precise than just taking the average income of a person

Reply:
For the testing part, we need to update the population file after this change is merged. Then we can set the income dependent thing to be true by default. 

For the data, I will try to find a German data and add it to the comment next to the script.